jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/ChangeConnectionSettingsMenu.kt

```diff
@@ -59,22 +59,27 @@ class ChangeAccountSettingsActionGroup(project: Project, private val mode: Chang
             val usedCredentials = accountSettingsManager.recentlyUsedCredentials()
             if (usedCredentials.isEmpty()) {
                 actions.add(Separator.create(message("settings.credentials")))
-
-                credentialSelector.isPopup = false
                 actions.add(credentialSelector)
             } else {
                 actions.add(Separator.create(message("settings.credentials.recent")))
                 usedCredentials.forEach {
                     actions.add(ChangeCredentialsAction(it))
                 }
 
-                credentialSelector.isPopup = true
-                actions.add(credentialSelector)
+                val allCredentials = DefaultActionGroup(message("settings.credentials.profile_sub_menu"), true)
+                allCredentials.add(credentialSelector)
+                allCredentials.add(Separator.create())
+                allCredentials.add(ActionManager.getInstance().getAction("aws.settings.upsertCredentials"))
+
+                actions.add(allCredentials)
             }
         }
 
-        actions.add(Separator.create())
-        actions.addAll(accountSettingsManager.connectionState.actions)
+        // Both mode == status bar version
+        if (mode == BOTH) {
```

Comment:
Should prly rename this mode at some point - cos you're right it is just the "StatusBar" version.